https://savannah.nongnu.org/bugs/index.php?48776
--- avrdude/usbtiny.c (revision 1395)
+++ avrdude/usbtiny.c (working copy)
@@ -455,6 +455,10 @@

   for (; addr < maxaddr; addr += chunk) {
     chunk = PDATA(pgm)->chunk_size;         // start with the maximum chunk size possible
+   
+    if (chunk > maxaddr - addr) {
+        chunk = maxaddr - addr;
+    }

     // Send the chunk of data to the USBtiny with the function we want
     // to perform
@@ -510,6 +514,10 @@
   for (; addr < maxaddr; addr += chunk) {
     // start with the max chunk size
     chunk = PDATA(pgm)->chunk_size;
+   
+    if (chunk > maxaddr - addr) {
+        chunk = maxaddr - addr;
+    }

     // we can only write a page at a time anyways
     if (m->paged && chunk > page_size) 
